### ISSUE TITLE
Non-existing TAGS file breaks autocompletion

### ISSUE THREAD

**Issue by vrom911** [syl20bnr/spacemacs#10319]
#### Description :octocat:

I'm continuing to see the following error after I've deleted `some-project`:

```
Company: backend (company-dabbrev-code company-gtags company-etags company-keywords) user-error: File ~/some-project/TAGS does not exist
```

I've deleted this project directory but currently I'm working on other project which is no related to `some-project`. I don't understand why it cares about `TAGS` file for other project.

#### Reproduction guide :beetle:
- Start Emacs
- Create two projects
- Generate TAGS file in one project
- Open other project
- Delete directory with first project
- Rerun spacemacs again and open buffer in another project

*Observed behaviour:* :eyes: :broken_heart:

Auto-completion is not working.

*Expected behaviour:* :heart: :smile:

Auto-completion is working.

#### System Info :computer:
- OS: gnu/linux
- Emacs: 25.2.2
- Spacemacs: 0.200.10
- Spacemacs branch: master (rev. 4bb4cb46)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(javascript html yaml no-dots git
            (version-control :variables version-control-diff-tool 'diff-hl version-control-diff-side 'left version-control-global-margin t)
            auto-completion spell-checking
            (haskell :variables haskell-completion-backend 'company-intero)
            (markdown :variables markdown-live-preview-engine 'vmd)
            helm emacs-lisp)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GSETTINGS NOTIFY ACL LIBSELINUX GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11

**Comment by CeleritasCelery** [comment]
You should report this upstream at company-mode

**Comment by dgutov** [comment]
Hmm, which version of company-mode was this?

I think I fixed it on 2018-01-11, later released in 0.9.5.

**Comment by ChShersh** [comment]
@dgutov I'm running into the same issue. Could you please help me with this as well? I can't figure out how can I check the version of package in Spacemacs :disappointed: Somehow this simple information is extremely hard to google and find out :confused:

**Comment by duianto** [comment]
@ChShersh HELM Describe Package, is available by pressing: `SPC h d p`

**Comment by ChShersh** [comment]
@duianto Thanks a lot :heart: 

@dgutov I have `company` version `0.9.6` and I still see this problem :disappointed:

**Comment by smonff** [comment]
I confirm this issue is not specific to Spacemacs: I use Prelude and got the same.

**Comment by dgutov** [comment]
Spacemacs and Prelude likely have some customization in common.

**Comment by smonff** [comment]
I managed to get company-mode working again after similar symptoms by flushing  `.emacs.desktop` and `.emacs.desktop.lock` from `.emacs.d` but I guess changing the value of `tags-file-name` with `(setq tags-file-name nil)` in `.emacs.desktop` would have been enough since it contained the value of what made company crash.

I know it should be reported to `company-mode`, but since Google took me there, I guess it could be useful for later readers.

**Comment by dgutov** [comment]
> Do you use (desktop-save-mode 1) ?

I don't.

Either way, a proper bug report would contain a step-by-step reproduction scenario. Preferably without Spacemacs (if you want to report that to company-mode).

**Comment by brymit** [comment]
I had the same exact issue and was able to fix it based on what @smonff and @dgutov said: the desktop might be causing the problems. I use bookmark+ instead of desktop but it also saves desktop-like files. Inspection verified that my .xyz.desktop file had the following lines:

```lisp
(setq tags-file-name "...foo/DbUiTest2/TAGS")
(setq tags-table-list '("...Someproject/TAGS"))
```

Removing these from the bookmark+ desktop file solved the issue.

(I hope I didn't do the wrong thing by commenting on this "stale" issue. I believe my confirmation indicates the issue can be safely closed.)

**Comment by lebensterben** [comment]
@brymit Is this still a problem on develop branch?

**Comment by brymit** [comment]
> 
> 
> @brymit Is this still a problem on develop branch?

I haven't tested that. In truth, I use emacs and not spacemacs. I considered not commenting since I don't use spacemacs, but I did comment because I hoped I was confirming that this issue is unrelated to spacemacs or company-mode, it is entirely in the user's environment, specifically the use of desktop or something similar like bookmark+. I apologize if I violated protocol or was not helpful.

The comments in this issue were extremely helpful to me because they reminded me that desktop/bookmark+ stores things that can cause problems later. And indeed that was the case here.
